gnosis/safe-react
# fix: reduce infura requests

## What it solves
Unnecessary Infura requests

## How this PR fixes it
Main investigation of our high-polled Infura requests showed that they are coming from our provider watcher (`eth_chainId`), Onboard (`eth_getBalance`) and WalletConnect (`eth_blockNumber`/`eth_getBlockByNumber`).

- [x] `eth_blockNumber`
- [x] `eth_getBalance`
- [x] `eth_getBlockByNumber`
- [x] `eth_chainId`

The base branch of this (the provider watcher cleanup) removed the 2 second interval polling for `ethChainId` and `ethAccounts`.

This removes the `eth_blockNumber`/`eth_getBlockByNumber` over-polling by [reducing the `blockPollingInterval` of WalletConnect](https://github.com/blocknative/onboard/blob/ba6f601cb173f33e258433cd65fcbde8faf30776/src/modules/select/wallets/wallet-connect.ts#L59) (via a custom WC module). This does not seem to have an adverse affect on WC, as it is apparently [a remnant from legacy `web3` configuration](https://github.com/WalletConnect/walletconnect-monorepo/issues/357#issuecomment-789663540).

The overpolling of `eth_getBalance` stems from an issue somewhere in the depency tree of the balance `stateSyncer` in the Onboard's WC module. I'The modified WC module also has no balance `stateSyncer`. We can't retrieve the balance via Onboard, but this is never used in the Safe.

## Links
- [Thread](https://gnosisinc.slack.com/archives/CQB729LLF/p1644489369402419) outlining the the high number of Infura requests.

## How to test it

- Connect to MM via Onboard:
  - `eth_blockNumber`/`eth_getBlockByNumber` - there should be no continuous polling per 4s for these
  - `eth_chainId` - there should be no continuous polling per 2s for this

- Connect to WC via Onboard:
  - `eth_getBalance` - there should be no continuous polling per second for this
